astropy/table/tests/test_operations.py

```diff
@@ -675,7 +689,7 @@ def test_table_meta_merge_conflict(self, operation_table_type):
 
     def test_bad_input_type(self, operation_table_type):
         self._setup(operation_table_type)
-        with pytest.raises(TypeError):
+        with pytest.raises(ValueError):
```

Comment:
Sigh, seeing this now in the tests makes me realize it's an API change.  Maybe it's better to just leave that exception (for empty input) as `TypeError` even though, technically, the input has the right type but has the wrong value?  Alternately just add this PR to the API change section of the changelog.

Reply:
yes, I've stuck with this a bit, too. Would you think keeping the API is more valuable here, or rather prefer the change?

Reply:
I was hoping you would decide.  :smile:  Anyway, I suppose that in practice it won't affect many (any) people, so maybe just do the API cleanup (so stick with `ValueError`) and put this in the changelog accordingly as an API change.